### ISSUE TITLE
[Bug]: Delete Account should delete uploaded media (pictures) too

### ISSUE THREAD

**Issue by maltfield** [LemmyNet/lemmy#4433]
### Requirements

- [X] Is this a bug report? For questions or discussions use https://lemmy.ml/c/lemmy_support
- [X] Did you check to see if this issue already exists?
- [X] Is this only a single bug? Do not put multiple bugs in one issue.
- [X] Is this a backend issue? Use the [lemmy-ui](https://github.com/LemmyNet/lemmy-ui) repo for UI / frontend issues.

### Summary

Bug: When a user deletes their account in Lemmy, they're told that it will "permanently delete all of your data from this instance" but, in fact, it doesn't delete all of their data. For example, the photos that they uploaded will not be deleted.

When a user deletes their account in Lemmy, they're presented with the following message

```
Warning: this will permanently delete all of your data from this instance
```

Users should expect then, that all of their data was deleted. But this is not the case for photos.

This means that all instance admins are violating the laws of GDPR for any users that live in the European Union.

### Steps to Reproduce

1. Create a new account on lemmy
2. Upload a picture to the instance (eg by creating a post and entering an image to upload)
3. Save the URL of the uploaded image
4. Delete the lemmy account
5. Confirm that the user no longer exists (eg visiting their old profile page should return a 500 error)
6. Confirm the uploaded image is not deleted (eg in a cache-cleared web browser, visit the URL of the uploaded image)

### Technical Details

Linux, Firefox

### Version

BE 0.19.3

**Comment by maltfield** [comment]
See also this ticket to add a section to the [Lemmy UI for users to be able to delete individual images](https://github.com/LemmyNet/lemmy-ui/issues/2359) without having to delete their account

 * https://github.com/LemmyNet/lemmy-ui/issues/2359

**Comment by maltfield** [comment]
@Nutomic I'm ok to cleanup the title of this ticket from being too unweildy, but I do think it's important to track which tickets are related to legal concerns of data privacy laws, such as GDPR.

Can we please add a tag to this issue, such as:

1. `gdpr` or
2. `legal`

**Comment by Nutomic** [comment]
I dont believe that GDPR applies to Lemmy unless it is provided as a commercial service.

https://www.twipla.com/en/blog/what-does-gdpr-compliance-mean-for-personal-websites

https://www.privacy-regulation.eu/en/recital-18-GDPR.htm

**Comment by maltfield** [comment]
EU residents' data protection laws apply to all websites, not just commercial ones.

The exemptions cited for "personal websites" do not apply to public Lemmy instances. Most lemmy instances Lemmy are *not* a "personal website" (unless it's a single user's personal instance).

It's extremely alarming if the lead lemmy devs think they're immune to criminal data protection practices.

**Comment by dessalines** [comment]
> Anyway, legality aside, the Lemmy project has a moral responsibility to implement software that ensures its users' [right to be forgotten](https://en.wikipedia.org/wiki/Right_to_be_forgotten) and other data privacy rights. Is the Lemmy team really deflecting and de-prioritizing data protection ethics?

Please do not engage with developers in this way. We have way more work than we can handle, and years of priorities before this. If its a priority for you, then work on it and we'll accept a PR.

**Comment by maltfield** [comment]
I appreciate the work you've done in writing lemmy, but please prioritize pending issues of criminal data processing.

> We have...years of priorities before this

This is the problem that I'm trying to resolve. Your priorities.

GPL provides legal protections to the developers writing the software; it does not protect the instances running the software.

By not prioritizing these issues, you're throwing users under the bus. By releasing a feature that allows users to upload content before implementing a means for them to delete that content, you're tying your community of instance admins to the railroad tracks and walking away.

**Comment by dessalines** [comment]
> This is the problem that I'm trying to resolve. Your priorities.

Would you mind if we set some of *your* priorities also? You're asking us to do free labor for you, that you're unwilling to do yourself. Do not put ultimatums and demands on people making FOSS, or I won't hesitate to block you from these repos.

**Comment by maltfield** [comment]
> Would you mind if we set some of your priorities also? 

I do not mind, please do.

In the FOSS projects that I lead, I largely let the priories be dictated by the community.

**Comment by dessalines** [comment]
Very well, your first priority is to learn rust and take on this issue, I'll assign it to you. Since you're okay with other people setting your priorities, this shouldn't be a problem. We'd much appreciate your help.

**Comment by maltfield** [comment]
For additional context of this issue, please see [Nightmare on Lemmy Street (A Fediverse GDPR Horror Story)](https://tech.michaelaltfield.net/2024/03/04/lemmy-fediverse-gdpr/)

[![Nightmare on Lemmy St - A GDPR Horror Story](https://tech.michaelaltfield.net/wp-content/uploads/sites/5/lemmy-fediverse-gdpr_featuredImage2.jpg)](https://tech.michaelaltfield.net/2024/03/04/lemmy-fediverse-gdpr/)

**Comment by twkonefal** [comment]
> Very well, your first priority is to learn rust and take on this issue, I'll assign it to you. Since you're okay with other people setting your priorities, this shouldn't be a problem. We'd much appreciate your help.

It's not his project and he's not its developer. You understand how bug reporting works, right?

Everyone, please calm down and focus on addressing the actual reported bug.

**Comment by korewaChino** [comment]
IMO, if you want something, DIY.

If you want a new feature or fix a bug in Lemmy, learn Rust and submit a patch. We'll help review your code if you did.

I agree that this violates GDPR, but FOSS developers, ***work on what they are interested in,*** not on what they have to do.

Honestly, I might be willing to work on a patch for this. But I'm a little tired right now, so if anyone else wants to work on this instead, go ahead. I'm a bit too busy with maintaining my day job code atm

**Comment by deanishe** [comment]
> but FOSS developers, work on what they are interested in

This being the case, I've submitted a PR to add a warning to the README regarding the non-compliance.

If nobody wants to write the code, fair enough. But people deserve to be warned that they could potentially be getting themselves into a lot of trouble for hosting Lemmy, imo.
